pkg/monitoring/prometheusagent/secret.go

```diff
@@ -63,21 +101,13 @@ func (pas PrometheusAgentService) buildRemoteWriteSecret(
 	}, nil
 }
 
-func readRemoteWritePasswordFromSecret(secret corev1.Secret) (string, error) {
-	remoteWriteConfig := RemoteWriteConfig{}
-	err := yaml.Unmarshal(secret.Data["values"], &remoteWriteConfig)
+func readMimirAuthPasswordFromSecret(secret corev1.Secret) (string, error) {
+	var secretData string
+
+	err := yaml.Unmarshal(secret.Data["credentials"], &secretData)
```

Comment:
We should check for existence of the `credentials` key here otherwise this would panic.

Reply:
Agreed